pheget/static/js/searchbox.js

```diff
@@ -1,60 +1,113 @@
-var getOmni = function (omniurl, callback) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', omniurl, true);
-    xhr.responseType = 'json';
-    xhr.onload = function () {
-        var status = xhr.status;
-        if (status === 200) {
-            callback(null, xhr.response);
-        } else {
-            callback(status, xhr.response);
-        }
-    };
-    xhr.send();
-};
+// Generic failure callback function
+function failureCallback(error) {
+    console.log('Error status: ' + error);
+}
 
-// eslint-disable-next-line no-unused-vars
-function parseSearch(searchText) {
+
+// Returns the data from an omnisearch fetch if successful, return nothing otherwise
+function getOmniSearch(searchText) {
+    const omniurl = `https://portaldev.sph.umich.edu/api/v1/annotation/omnisearch/?q=${searchText}&build=GRCh38`;
+    var data = fetch(omniurl)
+        .then((response) => response.json())
```

Comment:
As above, remember to check for `resp.ok` when using fetch.

Reply:
Added error handling for this.